drupal/sites/default/boinc/modules/boincwork/includes/boincwork.helpers.inc

```diff
@@ -1829,9 +1831,43 @@ function delete_col(&$array, $offset) {
 
 /**
  * Pretty print the application version
+ *
+ * Parameters are obtained from the BOINC project database, result,
+ * app, app_version, and platform tables.
+ *
+ * Parameters
+ * @param int $appverid
+ *  app_version_id field, may be negative- anonymous platforms.
+ * @param int $versnum
+ *  version_num field, application version number, may be NULL
+ * @param int $plan_class
+ *  plan_class field, applcation plan class, may be NULL
+ * @param string $platform
+ *  platform name, may be NULL
  */
-function pretty_application_version($vernum, $plan_class, $plfm) {
-  $prettyv = sprintf("%d.%02d", $vernum/100, $vernum%100);
-  $prettyc = ($plan_class) ? "($av->plan_class)" : '';
-  return "v$prettyv $prettyc $plfm";
-}
\ No newline at end of file
+function pretty_application_version($appverid, $vernum, $plan_class, $plfm) {
+  switch ($appverid) {
+    case ANON_PLATFORM_UNKNOWN:
+      return "Anonymous platform";
+      break;
+    case ANON_PLATFORM_CPU:
+      return "Anonymous platform CPU";
+      break;
+    case ANON_PLATFORM_NVIDIA:
+      return "Anonymous platform NVIDIA GPU";
+      break;
+    case ANON_PLATFORM_ATI:
+      return "Anonymous platform ATI GPU";
+      break;
+    case ANON_PLATFORM_INTEL:
+      return "Anonymous platform Intel GPU";
+      break;
+    case 0:
+      return "---";
+      break;
+    default:
```

Comment:
What exactly do you mean by first? The current switch-case/if-else tree is unambiguous for all defined `ANON*` values, 0 and *other* negative values. If you mean by "first" that the `if` should precede the `switch`, then no, that would in fact be wrong because the `ANON*` values are negative as well and thus need to be checked first.

tl;dr: I approve the change as it is